store/state/errors.go

```diff
@@ -0,0 +1,64 @@
+package state
+
+import (
+	"fmt"
+)
+
+// GetSafeErrorType is enum for indicating the type of error
+type GetSafeErrorType byte
+
+const (
+	// ErrTypeEmptyValue is used for nil byteslice values
+	ErrTypeEmptyValue GetSafeErrorType = iota
+	// ErrTypeUnmarshal is used for undeserializable values
+	ErrTypeUnmarshal
+)
+
+// Implements Formatter
+func (ty GetSafeErrorType) Format(msg string) (res string) {
+	switch ty {
+	case ErrTypeEmptyValue:
+		res = fmt.Sprintf("Empty Value found")
+	case ErrTypeUnmarshal:
+		res = fmt.Sprintf("Error while unmarshal")
+	default:
+		panic("Unknown error type")
+	}
+
+	if msg != "" {
+		res = fmt.Sprintf("%s: %s", res, msg)
+	}
+
+	return
+}
+
+// GetSafeError is error type for GetSafe method
+type GetSafeError struct {
```

Comment:
```suggestion
type SafeError struct {
```

Reply:
I really don't understand why we need an error wrapper type like this one. What happens if `GetSafe` just returns an `error`? Why do we want to wrap it? All those answers should be specified on a comment here